components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java

```diff
@@ -387,4 +392,36 @@ default boolean doPreUpdateApplicationTemplate(ServiceProvider serviceProvider,
 
         return true;
     }
+
+    /**
+     * Define any additional actions to the get all local authenticator method.
+     *
+     * @param tenantDomain The requested tenant domain.
+     * @param localAuthenticators   The local authenticators to be returned by the service.
+     */
+    default void doPostGetAllLocalAuthenticators(String tenantDomain,
```

Comment:
This listener should only be related to actions related to the application.
Filtering metadata using listeners doesn't look right.

Reply:
what is the purpose of introducing these listeners?

Reply:
This is to give extendability to do perform any operation on the returned local authenticators. The main use case will be to filter out any authenticators per tenant.